File: library/src/main/java/me/proxer/library/api/PagingLimitEndpoint.java
package me.proxer.library.api;

import javax.annotation.Nullable;

/**
 * Interface for classes which implement both {@link PagingEndpoint} and {@link LimitEndpoint}.
 *
 * @author Ruben Gees
 */
public interface PagingLimitEndpoint<T> extends PagingEndpoint<T>, LimitEndpoint<T> {

    /**
     * {@inheritDoc}
     */
    @Override
    PagingLimitEndpoint<T> page(@Nullable Integer page);

    /**
     * {@inheritDoc}
     */
    @Override
    PagingLimitEndpoint<T> limit(@Nullable Integer limit);
}

File: library/src/main/kotlin/me/proxer/library/api/PagingLimitEndpoint.kt
package me.proxer.library.api

/**
 * Interface for classes which implement both [PagingEndpoint] and [LimitEndpoint].
 *
 * @param <T> The type this endpoint returns.
 *
 * @author Ruben Gees
 */
interface PagingLimitEndpoint<T> : PagingEndpoint<T>, LimitEndpoint<T> {

    override fun page(page: Int?): PagingLimitEndpoint<T>
    override fun limit(limit: Int?): PagingLimitEndpoint<T>
}
